presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -0,0 +1,727 @@
+            inputsToConjuncts.computeIfAbsent(extractInputs(conjunct), k -> new ArrayList<>()).add(conjunct);
+        }
+
+        inputsToConjuncts.values().stream()
+                .map(expressions -> binaryExpression(AND, expressions))
+                .map(predicate -> new FilterFunction(session.getSqlFunctionProperties(), determinismEvaluator.isDeterministic(predicate), predicateCompiler.compilePredicate(session.getSqlFunctionProperties(), session.getSessionFunctions(), predicate).get()))
+                .forEach(filterFunctions::add);
+
+        return filterFunctions.build();
+    }
+
+    private static Set<Integer> extractInputs(RowExpression expression)
+    {
+        ImmutableSet.Builder<Integer> inputs = ImmutableSet.builder();
+        expression.accept(new InputReferenceBuilderVisitor(), inputs);
+        return inputs.build();
+    }
+
+    private static class InputReferenceBuilderVisitor
+            extends DefaultRowExpressionTraversalVisitor<ImmutableSet.Builder<Integer>>
+    {
+        @Override
+        public Void visitInputReference(InputReferenceExpression input, ImmutableSet.Builder<Integer> builder)
+        {
+            builder.add(input.getField());
+            return null;
+        }
+    }
+
+    private static String splitError(Throwable t, Path path, long start, long length)
+    {
+        return format("Error opening Hive split %s (offset=%s, length=%s): %s", path, start, length, t.getMessage());
+    }
+
+    private static class BucketAdapter
+            implements Predicate
+    {
+        public final int[] bucketColumns;
+        public final int bucketToKeep;
+        public final int tableBucketCount;
+        public final int partitionBucketCount; // for sanity check only
+        private final List<TypeInfo> typeInfoList;
+
+        public BucketAdapter(int[] bucketColumnIndices, List<HiveType> bucketColumnHiveTypes, int tableBucketCount, int partitionBucketCount, int bucketToKeep)
+        {
+            this.bucketColumns = requireNonNull(bucketColumnIndices, "bucketColumnIndices is null");
+            this.bucketToKeep = bucketToKeep;
+            this.typeInfoList = requireNonNull(bucketColumnHiveTypes, "bucketColumnHiveTypes is null").stream()
+                    .map(HiveType::getTypeInfo)
+                    .collect(toImmutableList());
+            this.tableBucketCount = tableBucketCount;
+            this.partitionBucketCount = partitionBucketCount;
+        }
+
+        @Override
+        public int[] getInputChannels()
+        {
+            return bucketColumns;
+        }
+
+        @Override
+        public boolean evaluate(SqlFunctionProperties properties, Page page, int position)
+        {
+            int bucket = getHiveBucket(tableBucketCount, typeInfoList, page, position);
+            if ((bucket - bucketToKeep) % partitionBucketCount != 0) {
+                throw new PrestoException(HIVE_INVALID_BUCKET_FILES, format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;%&#39; expression, bucket (declared at line 717), partitionBucketCount. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 719. Consider formatting this data with &#39;%d&#39; instead.